airflow-ctl/src/airflowctl/api/client.py

```diff
@@ -214,6 +231,79 @@ def __init__(
             **kwargs,
         )
 
+    def _clean_empty_values(self, data: dict[str, Any]) -> dict[str, Any]:
+        """
+        Recursively remove keys with None or empty values from a dictionary if they are not required in the datamodels.
+
+        Args:
+            data (dict): The input dictionary to clean.
+        Returns:
+            dict: The cleaned dictionary with empty values removed.
+        """
+        if not self.datamodel:
+            return data
+        # Check each datamodel and clean if field is nullable
+        cleaned_data = {}
+        for datamodel in self.datamodel:
+            # Create map from field name and required status
+            required_fields = []
+            for field_name, field_info in datamodel.__pydantic_fields__.items():
```

Comment:
Normally I would have just looked at `datamodel.model_fields.items()` -- does that not work here?